### DISCUSSION TITLE
[@types/node]: incorrect type for fs/promises/open

### DISCUSSION THREAD

**Discussion by jzabinski-dolios** [DefinitelyTyped/DefinitelyTyped#56994] [Issues with a @types package]
The signature for `open` is `open(path: PathLike, flags: string | number, mode?: Mode): Promise<FileHandle>`. `flags` was [changed](https://nodejs.org/docs/latest-v16.x/api/fs.html#fspromisesopenpath-flags-mode) in v11.1 to be optional and default to 'r'. The signature should be `open(path: PathLike, flags: string | number = 'r', mode?: Mode): Promise<FileHandle>`.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
this is unfortunate limitaton in TS declarations, default values for params are not supported, the `= r` is always dropped. So did you mean:
```ts
open(path: PathLike, flags?: string | number, mode?: Mode): Promise<FileHandle>
```

**Reply by jzabinski-dolios** [reply]
Yes, that makes sense. (Forgot we are dealing with just a declaration here.)
